### ISSUE TITLE
[Bug]: Narrator is not announcing collapsed state for “Elements must only use allowed ARIA attributes drop down” in “Accessibility” pane.

### ISSUE THREAD

**Issue by msftedad** [storybookjs/storybook#24126]
### Describe the bug

Narrator is not announcing collapsed state for “Elements must only use allowed ARIA attributes drop down” in “Accessibility” pane.

Actual Result:
While navigating to the “Elements must only use allowed ARIA attributes” drop down in Accessibility pane narrator is not announcing the collapsed state of the drop down.

Expected Result:
While invoking the “Elements must only use allowed ARIA attributes drop down” in Accessibility pane narrator should announce the collapsed state of the drop down.

### To Reproduce

1. Open URL: [Common Navigation - Scrolling Panel ⋅ Storybook (windows.net)](https://vivacommonstorybooktest.blob.core.windows.net/$web/index.html?path=/story/common-navigation--scrolling-panel&globals=backgrounds.grid:!false;backgrounds.value:transparent)
2. Navigate to “App bar” button at right side of the pane invoke it and navigate to “Default” link invoke it. 
3. Now navigate to “Accessibility” tab item invoke it and navigate to “Elements must only use allowed ARIA attributes drop down” and observe the narrator announcement.

Attachments:
[collapsed state .webm](https://github.com/storybookjs/storybook/assets/93735775/ec192fab-8781-407a-a703-6c6e5e2a9c05)

### System

```bash
OS: Windows 11
OS version: 22H2 (OS Build 22621.2134)
Edge Version: 116.0.1938.69 (Official build) (64-bit)
```

**Comment by Nevnet99** [comment]
I can take a look at this proposed solution would be to change the markup from a <div> with a role="button" so that we get the focus state without having to hack them onto a div although the tabIndex might get a bit wonky with this approach will try and get a PR up for this tomorrow 👍

**Comment by Sidnioulz** [comment]
Hi folks,

I've created an issue to track all four instances of this bug in one place as we need to decide if we solve them through a shared component or through local UI changes. I'm closing this as duplicate.

Thanks,
